### PR TITLE
Add availability-zone as optinal attribute for OCP cluster nodes, Fix #179

### LINKED ISSUE #179: Support for different availability zone for each type of OCP node
Current TF only supports the global availability zone, we want the TF can support different availability zone for different OCP node. For example master nodes using master availability zone, worker nodes for worker availability zone.

### PR DESCRIPTION
**Pull request by cs-zhang**
Fix 179

 Add `availability_zone` as the optional attribute for OCP node : bastion, bootstrap, master and worker. If it is not specified the `openstack_availability_zone` will be used for them.
 
Signed-off-by: CS Zhang <zhangcho@us.ibm.com>

### CODE REVIEW

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/178#" title="Author self-approved">cs-zhang</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **bpradipt** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @bpradipt` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervm).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervm/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: docs/var.tfvars-doc.md
Comment (yussufsh) [inline docs/var.tfvars-doc.md]: `availability_zone = ""` actually means default AZ if I am not wrong. So it is better to not provide "" in var.tfvars file or documentation IMO.

File: docs/var.tfvars-doc.md
Comment (yussufsh) [inline docs/var.tfvars-doc.md]: Removed additional space.

File: docs/var.tfvars-doc.md
Comment (yussufsh) [reply docs/var.tfvars-doc.md]: User can provide bastion.availability_zone if he need more customization. If only provided then we take the value.

File: var.tfvars
Comment (yussufsh) [inline var.tfvars]: Make it optional by removing it from here and keeping in the commented line below. SImilar to `fixed_ip_v4 `.

File: modules/4_nodes/nodes.tf
Comment (yussufsh) [inline modules/4_nodes/nodes.tf]: These conditions becomes simpler when we look for `is defined`.
```
availability_zone   = lookup(var.bootstrap, "availability_zone", var.openstack_availability_zone)
```

**Review by yussufsh — CHANGES_REQUESTED** [review]

**Comment by yussufsh** [comment]
@cs-zhang Please correct the typo in commit message and better describe the problem you are trying to fix in an [issue](https://github.com/ocp-power-automation/ocp4-upi-powervm/issues).

File: docs/var.tfvars-doc.md
Comment (cs-zhang) [reply docs/var.tfvars-doc.md]: Done, if the **availability_zone = ""** is defined, the default availability zone will be used.

File: docs/var.tfvars-doc.md
Comment (cs-zhang) [reply docs/var.tfvars-doc.md]: Done.

File: var.tfvars
Comment (cs-zhang) [reply var.tfvars]: Done.

File: modules/4_nodes/nodes.tf
Comment (cs-zhang) [reply modules/4_nodes/nodes.tf]: Done.

File: var.tfvars
Comment (yussufsh) [reply var.tfvars]: Can you please remove `availability_zone = ""` from here and other nodes config? Just add it to the commented line below.

Many users just edit and use this var.tfvars file. Changing this way will use default AZ for everyone if they don't notice. So better show it in commented lines.

Comment (cs-zhang) [reply var.tfvars]: Moved all to commented lines.

**Review by yussufsh — APPROVED** [review]
/lgtm

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/178#" title="Author self-approved">cs-zhang</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/178#pullrequestreview-604256955" title="LGTM">yussufsh</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervm).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervm/blob/master/OWNERS)~~ [yussufsh]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: docs/var.tfvars-doc.md
Comment (bpradipt) [inline docs/var.tfvars-doc.md]: nit: optinal => optional

Comment (bpradipt) [reply docs/var.tfvars-doc.md]: Also can you please include the `Fix #179` in the commit message.

**Comment by ltccci** [comment]
New changes are detected. LGTM label has been removed.

File: docs/var.tfvars-doc.md
Comment (cs-zhang) [reply docs/var.tfvars-doc.md]: Done

**Review by bpradipt — APPROVED** [review]
/lgtm

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/178#pullrequestreview-606150505" title="LGTM">bpradipt</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/178#" title="Author self-approved">cs-zhang</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/178#pullrequestreview-604256955" title="LGTM">yussufsh</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervm).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervm/blob/master/OWNERS)~~ [bpradipt,yussufsh]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by bpradipt — APPROVED** [review]

**Comment by ltccci** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/178#pullrequestreview-606151154" title="Approved">bpradipt</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/178#" title="Author self-approved">cs-zhang</a>*, *<a href="https://github.com/ocp-power-automation/ocp4-upi-powervm/pull/178#pullrequestreview-604256955" title="LGTM">yussufsh</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=ocp-power-automation%2Focp4-upi-powervm).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/ocp-power-automation/ocp4-upi-powervm/blob/master/OWNERS)~~ [bpradipt,yussufsh]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
